llvm/cmake/modules/AddLLVM.cmake

```diff
@@ -211,10 +211,10 @@ if (NOT DEFINED LLVM_LINKER_DETECTED AND NOT WIN32)
     )
 
   if(APPLE)
-    if("${stderr}" MATCHES "PROJECT:ld64")
+    if("${stderr}" MATCHES "PROGRAM:ld")
```

Comment:
I'm not certain I understand -- isn't this patch missing something so that it keeps working on pre Xcode 15 toolchains? Libc++ supports Xcode 15 only, but the rest of LLVM probably supports older versions as well?

Reply:
Yes, that's the intent -- in fact we are not checking the `PROJECT` part anymore, but the `PROGRAM` one, which I found empirically remains more "stable" across Xcode versions (while not making the check needlessly complicated with regexes):

```
> .../Xcode14.3.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld -v
@(#)PROGRAM:ld  PROJECT:ld64-857.1
...
```

```
> .../Xcode15.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/ld -v
@(#)PROGRAM:ld  PROJECT:dyld-1022.1
...
```

Reply:
Ahhh, now this all makes sense! I hadn't noticed that you switched to `PROGRAM:ld`.